### ISSUE TITLE
error: nodeFromFileInfo: get named security info failed with: "Client doesn't have necessary privilege"

### ISSUE THREAD

**Issue by riomccloud** [restic/restic#5004]
Output of `restic version`
--------------------------
restic 0.17.0 compiled with go1.22.5 on windows/amd64

What backend/service did you use to store the repository?
---------------------------------------------------------
Local filesystem (NTFS)

Problem description / Steps to reproduce
----------------------------------------
Command: `cd /d "C:\origem" && restic -r "C:\destino" backup . --tag exemplo1 --exclude=desktop.ini --skip-if-unchanged`

Output:

```
using parent snapshot cba99a3c

Files:           2 new,     0 changed,     1 unmodified
Dirs:            0 new,     0 changed,     0 unmodified
Added to the repository: 20.288 KiB (15.372 KiB stored)

processed 3 files, 18.292 KiB in 0:00
snapshot 3faaf8f3 saved
error: nodeFromFileInfo pasta: get named security info failed with: O cliente não tem o privilégio necessário.
Warning: at least one source file could not be read
```

Translations:
`pasta` - "folder" in Portuguese, literally a folder called folder
`O cliente não tem o privilégio necessário` - "Client doesn't have necessary privilege"

Expected behavior
-----------------
Simply don't fail to get named security info.

Actual behavior
---------------
I still didn't understand what causes this, it seems to be random. When backing up files in local filesystem (Windows, NTFS), sometimes it gives "nodeFromFileInfo" errors, saying "The client doesn't have necessary privileges".

Do you have any idea what may have caused this?
-----------------------------------------------
No. For now, I didn't had problems with cloud backends through Rclone.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Restic, alongside Rclone, is one of the most incredible pieces of software I ever used. It vastly improved the ease of doing secure and stable backups. Please keep it up! <3

**Comment by konidev20** [comment]
This is related to the backup of the Security Descriptors in Windows. Looks like the code handles the error 

```
// GetSecurityDescriptor takes the path of the file and returns the SecurityDescriptor for the file.
// This needs admin permissions or SeBackupPrivilege for getting the full SD.
// If there are no admin permissions, only the current user's owner, group and DACL will be got.
```

the relevant privilege checks and this error seemed to be part of the checks. 

```
// isHandlePrivilegeNotHeldError checks if the error is ERROR_PRIVILEGE_NOT_HELD
func isHandlePrivilegeNotHeldError(err error) bool {
	// Use a type assertion to check if the error is of type syscall.Errno
	if errno, ok := err.(syscall.Errno); ok {
		// Compare the error code to the expected value
		return errno == windows.ERROR_PRIVILEGE_NOT_HELD
	}
	return false
}
```

I think there is a bug in the `sd_windows.go` in the `fs` package.

**Comment by MichaelEischer** [comment]
> I think there is a bug in the `sd_windows.go` in the `fs` package.

Well, the assumption so far was that it's always possible to read the information requested by `lowBackupSecurityFlags`. But apparently that's not always the case. The challenge will be finding out which part of that information can't be accessed.

@riomccloud When you open the file properties in explorer for the file in question, are you able to access the security tab? The should also be a button titled "extended" in the lower right corner. Is that folder owned by your user? Do the errors also show up when creating a new folder in some other location?

Which Windows version are you using?

**Comment by riomccloud** [comment]
@MichaelEischer Yep, the permissions are all fine. I have created folders in multiple places, the same error happens sometimes.
I'm using Windows 10 21H2.

**Comment by MichaelEischer** [comment]
> I have created folders in multiple places, the same error happens sometimes.

Please be a bit more specific what "sometimes" means. What happens when you backup some of the problematic files again? Anything noteworthy about the storage location? Is it a local disk or a network file system etc.?

> I'm using Windows 10 21H2.

That version is already out for support for non-enterprise versions.

**Comment by MichaelEischer** [comment]
> I said "sometimes" because the behavior is really inconsistent. There are times when trying again does back up the files correctly, but there are times the error persists.

That really sounds like a bug in Windows or some security software. Restic always reads the security information for a file in the same way. So the operation should either always fail or succeed. Intermittent behavior sounds like the problem is somewhere else. https://github.com/restic/restic/pull/4977 also turns that error into a warning the still backs up the file in question, just without the security descriptor.

> > That version is already out of support for non-enterprise versions.
> 
> I'm using an Enterprise version.

I have to correct myself. The enterprise versions are also out of support by now, see [link](https://learn.microsoft.com/en-us/lifecycle/products/windows-10-enterprise-and-education).

**Comment by MichaelEischer** [comment]
I've just encountered the same error on an up to date Windows 10 installation. For some reason it occurred on a folder. My guess is that there's a race condition somewhere.

**Comment by MichaelEischer** [comment]
I think I've found the problem. There was a race condition when reading security descriptors for the first few files, such that the fallback to lower privileges did not always work correctly, see https://github.com/restic/restic/pull/5019 for details. Now, that I know what's the problem, I'm quite surprised that nobody else has encountered the same problem.
